t/t-batch-transfer.sh

```diff
@@ -184,13 +231,17 @@ begin_test "batch transfers with ssh endpoint (git-lfs-transfer)"
   git add .gitattributes test.dat
   git commit -m "initial commit"
 
+  # On Windows we do not multiplex SSH connections by default, so we
+  # enforce their use in order to match other platforms' connection counts.
+  git config --global lfs.ssh.autoMultiplex true
```

Comment:
The `GetExeAndArgs()` function in our `ssh` package effectively [sets](https://github.com/git-lfs/git-lfs/blob/f979a7db3b11d77975bc241f2c5ae71b02457d2c/ssh/ssh.go#L154) the default value of `lfs.ssh.autoMultiplex` to be `false` and not `true` because Windows SSH clients (of which there are a few) don't always support multiplexing SSH sessions over a single connection, and may fail if they see an unrecognized control socket option.

There was a discussion of this concern in https://github.com/git-lfs/git-lfs/pull/5537#discussion_r1347862427, which led to the changes from that PR #5537 setting the default to `false` on Windows.  The conclusion at that time was to just let Windows users who had OpenSSH installed (via Git for Windows, for instance) set the configuration option to `true` manually if they wanted to use multiplexing.

I did note, while working on this PR, that the `lfs.ssh.autoMultiplex` [section](https://github.com/git-lfs/git-lfs/blob/f979a7db3b11d77975bc241f2c5ae71b02457d2c/docs/man/git-lfs-config.adoc?plain=1#L90-L94) in our `git-lfs-config(5)` manual page doesn't explain this difference in the default value on Windows.  I have some work-in-progress documentation updates and I put an edit to the manual page into that set of changes, but since you've reminded me again about the issue, I'll just put that commit into this PR.